pkg/service/backup/restore_worker_tables.go

```diff
@@ -32,8 +32,8 @@ func (w *tablesWorker) restore(ctx context.Context, run *RestoreRun, target Rest
 			}
 		}
 	}
-	// Restore files
-	for _, l := range target.Location {
+	// Restore locations in deterministic order
+	for _, l := range target.copiedAndSortedLocations() {
```

Comment:
Because we don't allow updating location parameter of restore task, I think that this is not required in this case.
Moreover, `copiedAndSortedLocations` is something that fits into `GetRestoreTarget` more instead of being called here.